src/nwb/NWBFile.cpp

```diff
@@ -131,7 +140,15 @@ Status NWBFile::createElectricalSeries(
   return Status::Success;
 }
 
-void NWBFile::stopRecording() {}
+Status NWBFile::startRecording()
+{
+  return io->startRecording();
+}
+
+void NWBFile::stopRecording()
+{
+  io->stopRecording();
+}
```

Comment:
Can you clarify why `startRecording`, `stopRecording` are part of `NWBFile` and not `NWBRecording`? Since `NWBRecording` manages the recording process, those functions seem to logically belong there (or maybe they are needed on both `NWBFile` and `NWBRecording`, depending on if `NWBFile` has to do anything special here?)

Reply:
I think currently the main reason it's part of `NWBFile` is because the `io` object is a private member variable of `NWBFile`. 

As you mentioned above, currently `NWBRecording::openFile` assumes a specific recording process and automatically sets that up, calling `NWBFile::startRecording` at the end. However I agree if we want `NWBRecording` to manage the recording process and want to allow more customizability for different datatypes, it makes sense to at least have `startRecording` on both.

Reply:
> I think currently the main reason it's part of `NWBFile` is because the `io` object is a private member variable of `NWBFile`.

Let's discuss the design for `NWBRecording`. I'm wondering whether the `io` should be owned by `NWBRecording`, but that'll depend on where we decide to draw the line in terms of what the responsibility of `NWBFile` and `NWBRecording` are. But let's deal with that in a separate PR.